src/sentry/taskworker/workerchild.py

```diff
@@ -0,0 +1,389 @@
+from types import FrameType
+from typing import Any
+
+# XXX: Don't import any modules that will import django here, do those within child_process
+import orjson
+import sentry_sdk
+from django.core.cache import cache
+from sentry_protos.taskbroker.v1.taskbroker_pb2 import (
+    TASK_ACTIVATION_STATUS_COMPLETE,
+    TASK_ACTIVATION_STATUS_FAILURE,
+    TASK_ACTIVATION_STATUS_RETRY,
+    TaskActivation,
+    TaskActivationStatus,
+)
+from sentry_sdk.consts import OP, SPANDATA, SPANSTATUS
+from sentry_sdk.crons import MonitorStatus, capture_checkin
+
+logger = logging.getLogger("sentry.taskworker.worker")
+
+AT_MOST_ONCE_TIMEOUT = 60 * 60 * 24  # 1 day
+
+
+class ProcessingDeadlineExceeded(BaseException):
+    pass
+
+
+@dataclasses.dataclass
+class ProcessingResult:
+    """Result structure from child processess to parent"""
+
+    task_id: str
+    status: TaskActivationStatus.ValueType
+
+
+def child_worker_init(process_type: str) -> None:
+    """
+    Configure django and load task modules for workers
+    Child worker processes are spawned and don't inherit db
+    connections or configuration from the parent process.
+    """
+    from django.conf import settings
+
+    from sentry.runner import configure
+
+    if process_type == "spawn":
+        configure()
+
+    for module in settings.TASKWORKER_IMPORTS:
+        __import__(module)
+
+
+@contextlib.contextmanager
+def timeout_alarm(
+    seconds: int, handler: Callable[[int, FrameType | None], None]
+) -> Generator[None]:
+    """
+    Context manager to handle SIGALRM handlers
+
+    To prevent tasks from consuming a worker forever, we set a timeout
+    alarm that will interrupt tasks that run longer than
+    their processing_deadline.
+    """
+    original = signal.signal(signal.SIGALRM, handler)
+    try:
+        signal.alarm(seconds)
+        yield
+    finally:
+        signal.alarm(0)
+        signal.signal(signal.SIGALRM, original)
+
+
+def get_at_most_once_key(namespace: str, taskname: str, task_id: str) -> str:
+    # tw:amo -> taskworker:at_most_once
+    return f"tw:amo:{namespace}:{taskname}:{task_id}"
+
+
+def child_process(
+    child_tasks: queue.Queue[TaskActivation],
+    processed_tasks: queue.Queue[ProcessingResult],
+    shutdown_event: Event,
+    max_task_count: int | None,
+    processing_pool_name: str,
+    process_type: str,
+) -> None:
+    """
+    The entrypoint for spawned worker children.
+
+    Any import that could pull in django needs to be put inside this functiona
+    and not the module root. If modules that include django are imported at
+    the module level the wrong django settings will be used.
+    """
+    child_worker_init(process_type)
+
+    from sentry.taskworker.registry import taskregistry
+    from sentry.taskworker.state import clear_current_task, current_task, set_current_task
+    from sentry.taskworker.task import Task
+    from sentry.utils import metrics
+    from sentry.utils.memory import track_memory_usage
```

Comment:
These imports pull in a chain of modules that result in django being loaded. Django will then attempt to configure `settings` using the information in `os.environ`. However, getsentry + sentry have _special_ bootstrapping logic that `configure()` does.

By moving these imports here, we can defer loading django modules until after `child_worker_init()` and `configure()` have been called. Another solution I considered was to detect that we are in a child process and call `child_worker_init()` at the module level but it felt more hacky than using functions + deferred imports.